src/types/dataSources/MetAPI.ts

```diff
@@ -0,0 +1,19 @@
+import { IDataSource, DataSourceID } from './DataSource';
+import { MetApiCompactAirTemperature, WEATHER_MET_API } from '../../queries/metApi';
+import { QueryResult, useQuery as _useQuery } from '@apollo/client';
+
+type MetAPIVariables = {
+    lat: number;
+    lon: number;
+};
+
+export class MetAPIDataSource implements IDataSource<MetApiCompactAirTemperature, MetAPIVariables> {
+    graphQLQuery = WEATHER_MET_API;
+    dataSourceId = DataSourceID.MET_API_FORECAST;
+
+    useQuery(variables: MetAPIVariables): QueryResult<MetApiCompactAirTemperature, MetAPIVariables> {
+        return _useQuery(this.graphQLQuery, {
+            variables,
+        });
+    }
+}
```

Comment:
Don't use classes. Use objects. Move `useQuery` to a function that accepts `IDataSource`.

Reply:
Functional. Get outta here with that OOP